scripts/gpu_support/nvidia/install_cuda_and_libraries.sh

```diff
@@ -0,0 +1,219 @@
+eula_accepted=0
+EASYSTACK_FILE=
+TEMP_DIR=
+
+# Parse command-line options
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        --help)
+            show_help
+            exit 0
+            ;;
+        --accept-cuda-eula)
+            eula_accepted=1
+            shift 1
+            ;;
+        -e|--easystack)
+            if [ -n "$2" ]; then
+                EASYSTACK_FILE="$2"
+                shift 2
+            else
+                echo "Error: Argument required for $1"
+                show_help
+                exit 1
+            fi
+            ;;
+        -t|--temp-dir)
+            if [ -n "$2" ]; then
+                TEMP_DIR="$2"
+                shift 2
+            else
+                echo "Error: Argument required for $1"
+                show_help
+                exit 1
+            fi
+            ;;
+        *)
+            show_help
+            fatal_error "Error: Unknown option: $1"
+            ;;
+    esac
+done
+
+if [[ -z "${EASYSTACK_FILE}" ]]; then
+    fatal_error "Need the name/path to an easystack file. See command line options\n"
+fi
+
+# Make sure EESSI is initialised
+check_eessi_initialised
+
+# As an installation location just use $EESSI_SOFTWARE_PATH but replacing `versions` with `host_injections`
+# (CUDA is a binary installation so no need to worry too much about the EasyBuild setup)
+export EESSI_SITE_INSTALL=${EESSI_SOFTWARE_PATH/versions/host_injections}
+
+# we need a directory we can use for temporary storage
+if [[ -z "${TEMP_DIR}" ]]; then
+    tmpdir=$(mktemp -d)
+else
+    mkdir -p ${TEMP_DIR}
+    tmpdir=$(mktemp -d --tmpdir=${TEMP_DIR} cuda_n_co.XXX)
+    if [[ ! -d "$tmpdir" ]] ; then
+        fatal_error "Could not create directory ${tmpdir}"
+    fi
+fi
+echo "Created temporary directory '${tmpdir}'"
+
+echo "MODULEPATH=${MODULEPATH}"
+echo "List available *CUDA* modules before loading EESSI-extend/${EESSI_VERSION}-easybuild"
+module avail CUDA
+
+# load EESSI-extend/2023.06-easybuild module && verify that it is loaded
+EESSI_EXTEND_MODULE="EESSI-extend/${EESSI_VERSION}-easybuild"
+module load ${EESSI_EXTEND_MODULE}
+ret=$?
+if [ "${ret}" -ne 0 ]; then
+    fatal_error "An error occured while trying to load ${EESSI_EXTEND_MODULE}\n"
+fi
+
+echo "MODULEPATH=${MODULEPATH}"
+echo "List available *CUDA* modules after loading EESSI-extend/${EESSI_VERSION}-easybuild"
+module avail CUDA
+
+# use install_path/modules/all as MODULEPATH
+SAVE_MODULEPATH=${MODULEPATH}
+MODULEPATH=${EASYBUILD_INSTALLPATH}/.modules/all
+echo "set MODULEPATH=${MODULEPATH}"
+
+# show EasyBuild configuration
+echo "Show EasyBuild configuration"
+eb --show-config
+
+# do a 'eb --dry-run-short' with the EASYSTACK_FILE and determine list of packages
+# to be installed
+echo ">> Determining if packages specified in ${EASYSTACK_FILE} are missing under ${EESSI_SITE_INSTALL}"
+eb_dry_run_short_out=${tmpdir}/eb_dry_run_short.out
+eb --dry-run-short --rebuild --easystack ${EASYSTACK_FILE} 2>&1 | tee ${eb_dry_run_short_out}
```

Comment:
With the changes above, do you even need `--rebuild`?

Reply:
Wait, that still doesn't explain why there's `--rebuild`. And there's no point anyway, since we then only grep `[ ]`, i.e. stuff that is _not_ present anyway. I'll remove it.